### ISSUE TITLE
BatchNorm is the first layer for ResNetV2 in model_zoo

### ISSUE THREAD

**Issue by vipinpillai** [dmlc/gluon-cv#873]
The resnet.py code in the model_zoo has BatchNorm as the first layer defined for all models using ResNetV2.
The line [here](https://github.com/dmlc/gluon-cv/blob/master/gluoncv/model_zoo/resnet.py#L461) in the class ResNetV2 adds norm_layer as the first layer for `self.features` after initializing it as `nn.HybridSequential`. Since the `hybrid_forward` function calls `self.features` for every input, we would always have this batch norm layer called first.

**Comment by hetong007** [comment]
Having BatchNorm as the first layer doesn't break anything, and it can be further seen as an automatic image augmentation mechanism.

**Comment by vipinpillai** [comment]
@hetong007 I don't think it's a good idea to keep it just because it doesn't break anything. I would expect the model architecture definition in the model_zoo to exactly match with the original implementation.
